scipy/stats/_stats.pyx

```diff
@@ -753,3 +753,87 @@ def gaussian_kernel_estimate(points, values, xi, precision, dtype, real _=0):
                 estimate[j, k] += values_[i, k] * arg
 
     return np.asarray(estimate)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def gaussian_kernel_estimate_log(points, values, xi, precision, dtype, real _=0):
+    """
+    def gaussian_kernel_estimate_log(points, real[:, :] values, xi, precision)
+    Evaluate the log of the estimated pdf on a provided set of points.
+    Parameters
+    ----------
+    points : array_like with shape (n, d)
+        Data points to estimate from in d dimensions.
+    values : real[:, :] with shape (n, p)
+        Multivariate values associated with the data points.
+    xi : array_like with shape (m, d)
+        Coordinates to evaluate the estimate at in d dimensions.
+    precision : array_like with shape (d, d)
+        Precision matrix for the Gaussian kernel.
+    Returns
+    -------
+    estimate : double[:, :] with shape (m, p)
+        The log of the multivariate Gaussian kernel estimate evaluated at the input coordinates.
+    """
+
+    # This implementation is derived from `gaussian_kernel_estimate`, huge thanks to the authors.
+
+    cdef:
+        real[:, :] points_, xi_, values_, log_values_
+        real[:, :] whitening, estimate, max_
+        real[:, :, :] estimate_
+        int i, j, k
+        int n, d, m, p
+        real arg, residual, log_arg, log_norm
+
+    n = points.shape[0]
+    d = points.shape[1]
+    m = xi.shape[0]
+    p = values.shape[1]
+
+    # Rescale the data
+    whitening = np.linalg.cholesky(precision).astype(dtype, copy=False)
+    points_ = np.dot(points, whitening).astype(dtype, copy=False)
+    xi_ = np.dot(xi, whitening).astype(dtype, copy=False)
+    values_ = values.astype(dtype, copy=False)
+
+    # Log transformation of values_. It is faster than np.log(values_).
+    log_values_ = np.zeros((n, p), dtype)
+    for i in range(n):
+        for k in range(p):
+            log_values_[i, k] = math.log(values_[i, k])
+
+    # Evaluate the normalisation
+    log_norm = (- d / 2) * math.log(2 * PI)
+    for i in range(d):
+        log_norm += math.log(whitening[i, i])
+
+    # Create the result array and evaluate the weighted sum
+    estimate = np.zeros((m, p), dtype)
+    max_ = np.zeros((m, p), dtype) - np.inf
+    estimate_ = np.zeros((m, p, n), dtype)
```

Comment:
Thank you. I changed them to `empty` and checked but consequently some values of the result array became `nan`. I'm not sure why.

Reply:
It's because of the `- np.inf` I suppose.

p.s. please don't resolve the conversations. It's better to let reviewers close them so they can keep track of things they want to look at.

Reply:
Actually yes, `max` is ok to be zeros because of the comparison. But `estimate_` has no need to be.

Reply:
Sorry for resolving the conversations. I reopened all of them. I'll leave it to you and reviewers to close them.

Reply:
For `estimate_`, I changed `zeros` to `empty` at [ccf299f](https://github.com/scipy/scipy/pull/15493/commits/ccf299feeb54c8517aa47736084029762119c005). Thanks.

Reply:
No problem, all communities work differently with conversations 😅